images/executor/README.md

```diff
@@ -0,0 +1,57 @@
+# Executor
+
+This image will be used to execute course code in isolation in production environments.
+
+## Release process
+
+This image is built and pushed to the container registry as `prairielearn/executor:GIT_HASH`, where `GIT_HASH` is the Git commit hash of the version of PrairieLearn the executor image was built from. This allows us to ensure a given version of PrairieLearn is always running with the correct executor image.
+
+There are scripts in the `/tools` directory that will aid with the building and releasing of this image. They will automatically determine the correct tag from the underlying Git repository. The scripts should be run from the root of the repository.
+
+To build and tag the image for local testing, run:
+
+```sh
+./tools/executor/build.js
+```
+
+To push the built image to the Docker and ECR registries, run:
+
+```
+./tools/executor/release.js
+```
+
+## In-dev testing
+
+Build the image using the above `build-executor.sh` script. Make note of the name of the resulting image; you'll need this image name momentarily.
+
+Start the container, using the version of the container that you noted above (the examples below use the Git hash `12345`). Note that we mount in the PrairieLearn `elements` directory so that the examples below can use PrairieLearn's elements:
+
+```sh
+docker run --rm -it -v /path/to/checked/out/PrairieLearn/elements/:/elements/ prairielearn/executor:12345
```

Comment:
Why do we mount over `/elements/` rather than `/PrairieLearn/elements/`?
```suggestion
docker run --rm -it -v /path/to/checked/out/PrairieLearn/elements/:/PrairieLearn/elements/ prairielearn/executor:12345
```

Reply:
Ah, good catch - this is no longer necessary since the executor runs on an image built on top of the `prairielearn/prairielearn` image. I'll update these instructions.